mobile/src/main/java/rs/readahead/washington/mobile/views/activity/viewer/SharedMediaFileViewModel.kt

```diff
@@ -0,0 +1,201 @@
+     *
+     * @param vaultFile The VaultFile to be deleted.
+     */
+    fun deleteMediaFiles(vaultFile: VaultFile?) {
+        disposables.add(MyApplication.rxVault.delete(vaultFile)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
+                // Notify observers that the deletion was successful, providing a boolean flag indicating success
+                { isDeleted: Boolean? -> _onMediaFileDeleted.postValue(isDeleted) }
+            ) { throwable: Throwable? ->
+                // Handle any errors that occurred during the deletion process
+                FirebaseCrashlytics.getInstance().recordException(throwable!!)
+                _error.postValue(R.string.gallery_toast_fail_deleting_files)
+
+            })
+    }
+
+
+    /**
+     * Confirms whether to delete the specified [vaultFile] based on its association with other FormMediaFiles.
+     *
+     * @param vaultFile The VaultFile to be confirmed for deletion.
+     */
+    fun confirmDeleteMediaFile(vaultFile: VaultFile) {
+        disposables.add(
+            keyDataSource.dataSource
+                .flatMapSingle { obj: DataSource -> obj.reportMediaFiles }
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ formMediaFileList: List<FormMediaFile> ->
+                    if (formMediaFileList.isEmpty()) {
+                        // Set confirmation based on the association with FormMediaFiles
+                        mediaFileDeleteConfirmation = MediaFileDeleteConfirmation(vaultFile, false)
+                        _onMediaFileDeleteConfirmed.postValue(mediaFileDeleteConfirmation)
+                    } else {
+                        var isShowConfirmation = false
+                        //#1. Iterate through list
+                        for (formMediaFile in formMediaFileList) {
+                            if (formMediaFile.id == vaultFile.id) {
+                                isShowConfirmation = true
+                                break
+                            }
+                        }
+                        mediaFileDeleteConfirmation =
+                            MediaFileDeleteConfirmation(vaultFile, isShowConfirmation)
+                        _onMediaFileDeleteConfirmed.postValue(mediaFileDeleteConfirmation)
+                    }
+                }, { throwable: Throwable? ->
+                    // Handle any errors that occurred during data retrieval
+                    FirebaseCrashlytics.getInstance().recordException(throwable!!)
+                    _error.postValue(R.string.gallery_toast_fail_deleting_files)
+                })
+        )
+    }
+
+    /**
+     * Retrieves a VaultFile based on its [id].
+     *
+     * @param id The ID of the VaultFile to retrieve.
+     */
+    fun getMediaFile(id: String?) {
+        disposables.add(
```

Comment:
Same here why the is nullable, let's have an early return, and apply the logic on the rest of the code please.